### PR TITLE
fix: fixed and updated grpc-out options (PR#231)

### PR DESCRIPTION
**Pull request by wespen**
Fix for grpc-out and Typescript code generation added in [PR#231](https://github.com/namely/docker-protoc/pull/231)
Relates to issue #230 

All options work properly now.

Fixes:
- Switched to [grpc_tools_node_protoc_ts](https://www.npmjs.com/package/grpc_tools_node_protoc_ts#how-to-use) from ts-protoc-gen
- Updated Node to v14. (latest LTS)
- Updated tests
- Extracted vars to variables.sh (extracted all others as well) for easier updating

### CODE REVIEW

File: all/test/include.proto [hunk 1]
```diff
@@ -2,10 +2,8 @@ syntax = "proto3";
 
 package Messages;
 
-option go_package = "all;all";
+option go_package = "all/test";
```
Comment (ido-namely) [inline all/test/include.proto:5]: Wondering why the go package name had to be changed?
iirc it's set this way to differentiate and test the source relative option.

File: all/test.sh [hunk 2]
```diff
@@ -247,3 +247,4 @@ if [[ ! -f gen/test.jar ]]; then
   echo "Expected gen/test.jar to be a jar file."
   exit 1
 fi
+rm -rf gen
```
Comment (ido-namely) [inline all/test.sh:250]: can you please add a new line here?

File: all/test/test.proto [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
 
+package Messages;
+
+import "all/test/include.proto";
 import "google/api/annotations.proto";
 import "google/protobuf/field_mask.proto";
-import "all/test/include.proto";
 
-package Messages;
+option go_package = "all/test";
```
Comment (ido-namely) [inline all/test/test.proto:9]: same comment as above

File: Dockerfile [hunk 4]
```diff
@@ -110,22 +117,30 @@ FROM debian:$debian-slim AS protoc-all
...
 
-# Add TypeScript support
+# Install latest Node version
+RUN curl -fsSL https://deb.nodesource.com/setup_${node_version}.x | bash -
+RUN apt-get install -y nodejs
 
+# Add TypeScript support
 RUN npm config set unsafe-perm true
-RUN npm i -g ts-protoc-gen@0.14.0
+RUN npm i -g grpc_tools_node_protoc_ts@$grpc_tools_node_protoc_ts_version grpc-tools@$grpc_tools_version protoc-gen-grpc-web@$protoc_gen_grpc_web_version
```
Comment (ido-namely) [inline Dockerfile:143]: is `grpc_tools`/`grpc_tools_version` specific to node/js?

File: variables.sh [hunk 5]
```diff
@@ -1,13 +1,20 @@
...
-BUILDS=("protoc-all" "protoc" "prototool" "grpc-cli" "gen-grpc-gateway")
+
+GO_VERSION=${GO_VERSION:-1.14}
+GRPC_VERSION=${GRPC_VERSION:-1.36}
+GRPC_JAVA_VERSION=${GRPC_JAVA_VERSION:-1.36}
+GRPC_WEB_VERSION=${GRPC_WEB_VERSION:-1.2.1}
+GRPC_GATEWAY_VERSION=${GRPC_GATEWAY_VERSION:-2.0.1}
+UBER_PROTOTOOL_VERSION=${UBER_PROTOTOOL_VERSION:-1.3.0}
+SCALA_PB_VERSION=${SCALA_PB_VERSION:-0.11.0}
+NODE_VERSION=${NODE_VERSION:-15}
```
Comment (nycdotnet) [inline variables.sh:17]: if there's any way we could stay on the latest LTS of Node, I'd appreciate it.  Today that's 14.x latest.

File: all/test/include.proto (hunk 1 above)
Comment (wespen) [reply all/test/include.proto:5]: This is the output of the protoc-gen-go
```
bash ./all/test.sh                       
Testing language go gen/pb-go --with-gateway
protoc-gen-go: invalid Go import path "all" for "all/test/include.proto"

The import path must contain at least one forward slash ('/') character.

See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.

--go_out: protoc-gen-go: Plugin failed with status code 1.
```
Details:
https://github.com/protocolbuffers/protobuf-go/blob/master/compiler/protogen/protogen.go#L259

Comment (ido-namely) [reply all/test/include.proto:5]: thanks, I realize protoc-gen-go has been upgraded lately.
from my local testing, I think changing the package to `option go_package = "all/;all";` in 
`all/test/include.proto` and `all/test/test.proto` would do.
Can you please try that?

Comment (wespen) [reply all/test/include.proto:5]: Sure. Ill test it out.

Comment (wespen) [reply all/test/include.proto]: Works, updated test and proto files.
Pushed.

**Comment by nycdotnet** [comment]
Thanks for this contribution - I'm going to review this today.

**Review by ido-namely — APPROVED** [review]
Thank you!

**Comment by juansalvatella** [comment]
I just stumbled across this issue and I was wondering when it will be merged. When should we expect a release with this? Or any tips on how I could try this myself?

**Comment by nycdotnet** [comment]
Should be soon - so sorry it's taken me a while to get to the QA

**Comment by nycdotnet** [comment]
Sorry this took so long - thank you for your contribution!!

**Review by nycdotnet — APPROVED** [review]

**Pull request merged**